docs/design/proposal/dependency-upgrades/aws-sdk-upgrades.md

```diff
@@ -0,0 +1,152 @@
+
+#### Current steps
+* At first ACK runtime gets updated with the newer version of aws-sdk-go and
+  a new patch release is made for ACK runtime.
+* Then ACK code-generator is updated with new ACK runtime release. Because of
+  transitive dependency the minimum required version of aws-sdk-go in 
+  code-generator gets updated with version in latest ACK runtime
+  release.
+* Then a new code-generator release is made, which triggers ACK service 
+  controller auto generation.
+* Controller auto-generation script updates the ACK runtime version in each 
+  controller to new ACK runtime version.
+* The update to new ACK runtime in service controller repositories updates the
+  minimum aws-sdk-go version in service controller go.mod file.
+* During auto-generation, ACK code-generator uses the aws-sdk-go present in
+  the service controller go.mod file to generate new service controller.
+
+The current process is automated for most parts but it misses the trigger to
+updating aws-sdk-go in ACK runtime and generating a preview of all service
+controller builds with new aws-sdk-go version.
+
+#### aws-sdk-go Releases
+* aws-sdk-go **v1** has two kinds of releases. **Minor version updates** which
+mainly reflect the changes in **SDK Features**(enabling FIPS endpoints) and
+**Patch version updates** which mainly reflect **Service Client Updates**(updates
+in specific AWS service API + documentation). See this
+[v1.42.0 changelog](https://github.com/aws/aws-sdk-go/releases/tag/v1.42.0) as an example
+
+## Scope
+This documentation will focus on automating the updates for aws-sdk-go minor version
+upgrades. Minor version upgrades will be done starting from ACK runtime to code-generator
+and then to all service-controllers using existing automation.
+
+Another [document](./aws-sdk-patch-version-upgrades.md) will focus on how to upgrade
+individual service controllers for the patch releases because Patch releases will only
+impact specific service controllers not the ACK runtime behavior (except IAM service
+changes)
+
+## High Level Overview
+
+The automation will be divided in following sub-parts
+
+### Trigger
+Trigger is responsible for deciding when to upgrade ACK core libraries and service
+controllers with the new aws-sdk-go version. The trigger can be event-driven(tracking
+a new release of aws-sdk-go using webhooks or other methods) or periodic.
+
+### Verification
+Verification is responsible for providing the preview of changes caused by
+aws-sdk-go upgrade. Example: Which controller(s) build successfully with the
+new aws-sdk-go version? Does a new resource/ new fields gets added as part
+of controller build? etc...
+
+### Release
+The actual process of upgrading the go.mod files with latest aws-sdk-go version
+and raising PR.
+
+## Implementation
+
+### Trigger
+For the trigger, the periodic job option should be preferred that will
+run every day. 24 hours frequency is good enough to not delay verifying
+the new aws-sdk-go release. 
```

Comment:
I would add a filter here to ignore any releases that are only patches - passing only minor version updates through to verification. There is no point going any further with testing past this point otherwise. This should be as simple as checking if the patch version of the SDK semver is 0. Whenever you update the minor version, you have to reset the patch to 0.